base/broadcast.jl

```diff
@@ -283,6 +288,15 @@ function broadcast_c(f, ::Type{Tuple}, As...)
     n = length(shape[1])
     return ntuple(k->f((_broadcast_getindex(A, k) for A in As)...), n)
 end
+function broadcast_c(f, ::Type{Nullable}, a...)
+    nullity = any(isnull, a)
```

Comment:
I am done (I think) with the scalar broadcast case, which now has optimal performance (as the generated code is exactly the same as handwritten code), and will work on the array broadcast case next.